### PR TITLE
Feature/friendshipv2

### PR DESCRIPTION
**Pull request by Mindo-Joseph**
### What does this PR do?
- Update the friendship controller to prevent duplication of created friendships

### How can this project be tested?
- Run ```rspec spec/features/friendships_spec.rb```

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -2,11 +2,21 @@
...
+          <%= link_to 'See Profile',  user_path(user), class: 'profile btn btn-outline-success btn-sm' %>
+        <% if signed_in? && !Friendship.reacted?(current_user.id, user.id) && current_user != user %>
+          <td> <%= link_to 'Add Friend', friendships_create_path(ids: {id1: current_user.id, id2: user.id}), class: 'friend btn btn-success btn-sm' %></td>
+        <% elsif signed_in? && !Friendship.confirmed_record?(current_user.id, user.id) && current_user != user %>
+          <% user.pending_friendships.each do |friendship| %>
+            <% if current_user %>
+              <%= link_to 'Cancel Request', friendships_destroy_path(friendship_id: friendship.id),  class: 'btn btn-danger btn-sm' %>
+              <% end %>
+          <% end %>
+        <% end %>
```
Comment (maelfosso) [inline app/views/users/index.html.erb]: That is what we call **logic in views**. I mean `if`, `if/else`, `if/elsif` and so on.
It should be implemented in **views helpers**

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -2,11 +2,21 @@
   <ul class="users-list">
     <% @users.each do |user| %>
       <li>
-        Name: <%= user.name %>
-        <span class="profile-link">
-          <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
-        </span>
+        <%= user.name %></br>
+          <%= link_to 'See Profile',  user_path(user), class: 'profile btn btn-outline-success btn-sm' %>
+        <% if signed_in? && !Friendship.reacted?(current_user.id, user.id) && current_user != user %>
```
Comment (maelfosso) [inline app/views/users/index.html.erb]: `Friendship.reacted?` is not good here.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+	
+	<br>
+	
+	<div class="card col-sm-9">
+	  <h3>Recent posts:</h3>
+       <% if signed_in? && Friendship.reacted?(current_user.id, @user.id) %>
+        <ul class="posts"><%= render @posts %></ul>
+	   <% elsif signed_in? && @user == current_user %>
+	     <ul class="posts"><%= render @user.posts %></ul>
+       <% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: Use **views helpers** please

File: app/views/shared/_user_posts.html.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<% if current_user.friends %>
+  <%= render @posts %>
+<% end %>
```
Comment (maelfosso) [inline app/views/shared/_user_posts.html.erb]: Use **views helpers**

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the effort put into this project so far.
Please, a bit more effort to move forward

1- In the user list, after sending a friendship request, I didn't see the two buttons for **accepting** or **rejecting** the friendship request in front of the user's name. They were only available once opening the user's profile. They must be available in the user's list

2- Look the Timeline from these two friends
**Alice Bar**
Screenshot at 2020-07-29 18:42:17
**Bob Bar**
Screenshot at 2020-07-29 18:42:39

You can notice from those two timelines are different.

3- Here is what we have from the console 
```rb
2.7.1 :001 > Friendship.all
  Friendship Load (0.4ms)  SELECT "friendships".* FROM "friendships" LIMIT $1  [["LIMIT", 11]]
 => #<ActiveRecord::Relation [#<Friendship id: 1, user_id: 2, friend_id: 1, confirmed: true, created_at: "2020-07-29 17:38:47", updated_at: "2020-07-29 17:41:00">]> 
2.7.1 :002 >
```
In this milestone, you're supposed to implement the two rows friendship. So for these 2 users, the table is supposed to have two rows.

4- Read the comments left below.

5- Update the **integration tests**

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by edieatha** [comment]
## TSE Recommendations Applied
* Remove logic from views with helpers and partials
* Update integration tests
* Add 2 records per pair of friendships

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team
### The pr branch has conflicts that you need to solve before the project can be reviewed
### Make the changes and submit for another code review. Good luck!

By Raymond
Slack @iamraymond

**Pull request merged**